src/Object/Contact.php

```diff
@@ -99,18 +100,12 @@ public static function markForArchival(array $contact)
 		}
 
 		if ($contact['term-date'] <= NULL_DATE) {
-			q(
-				"UPDATE `contact` SET `term-date` = '%s' WHERE `id` = %d", dbesc(datetime_convert()), intval($contact['id'])
-			);
+			dba::update('contact', array('term-date' => datetime_convert()), array('id' => $contact['id']));
 
 			if ($contact['url'] != '') {
-				q(
-					"UPDATE `contact` SET `term-date` = '%s'
-				WHERE `nurl` = '%s' AND `term-date` <= '1000-00-00'", dbesc(datetime_convert()), dbesc(normalise_link($contact['url']))
-				);
+				dba::update('contact', array('term-date' => datetime_convert()), array('nurl' => normalise_link($contact['url']), 'term-date' <= '1000-00-00'));
```

Comment:
Shouldn’t this be a constant? I know we have another constant for the smallest date-time in MySQL.

Reply:
'term-date' <= 'MIN(term-date)'  ?

Reply:
In the DB docs usually it is `0001-01-01 `

Reply:
In these instances, I was just trying to get into the new calls, I wasn't looking at changing any data on my own :smile:

Reply:
I guess this value is intended. We seem to have left it over when creating the constant ```NULL_DATE```

Reply:
Could it be a legacy leftover? If you of all people don't know why it's like that, I'd say let's use the existing constant.

Reply:
See here : https://github.com/friendica/friendica/blob/2ee928b0816094f3166c65b851bc7f6c84e9f3c9/src/Object/Contact.php#L161

We are using `NULL_DATE` for `term-date`, might as well use it everywhere which includes this line and line 152.